addon/-private/system/relationships/state/has-many.js

```diff
@@ -109,8 +111,46 @@ export default class ManyRelationship extends Relationship {
 
     assertPolymorphicType(this.internalModel, this.relationshipMeta, internalModel);
     super.addInternalModel(internalModel, idx);
-    // make lazy later
-    this.manyArray._addInternalModels([internalModel], idx);
+    this.scheduleManyArrayUpdate(internalModel, idx);
+  }
+
+  scheduleManyArrayUpdate(internalModel, idx) {
+    // ideally we would early exit here, but some tests
+    //   currently suggest that we cannot.
+    // if (!this._manyArray) {
+    //   return;
+    // }
+
+    let pending = this._pendingManyArrayUpdates = this._pendingManyArrayUpdates || [];
+    pending.push(internalModel, idx);
+
+    if (this._willUpdateManyArray === true) {
+      return;
+    }
+
+    this._willUpdateManyArray = true;
+    let backburner = this.store._backburner;
+
+    backburner.join(() => {
```

Comment:
IMHO, it would be nice to remove this (and rely on the auto-run). What do you think?